k3k-kubelet/kubelet.go

```diff
@@ -71,15 +78,51 @@ func newKubelet(ctx context.Context, c *config, logger *k3klog.Logger) (*kubelet
 	}
 
 	virtClient, err := kubernetes.NewForConfig(virtConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	hostMgr, err := ctrl.NewManager(hostConfig, manager.Options{
+		Scheme: baseScheme,
+		Metrics: ctrlserver.Options{
+			BindAddress: ":8083",
+		},
+		Cache: cache.Options{
+			DefaultNamespaces: map[string]cache.Config{
+				c.ClusterNamespace: {},
+			},
+		},
+	})
+	if err != nil {
+		fmt.Printf("unable to create controller-runtime mgr for host cluster: %s", err.Error())
+		os.Exit(-1)
+	}
+
+	virtualScheme := runtime.NewScheme()
+	// virtual client will only use core types (for now), no need to add anything other than the basics
+	err = clientgoscheme.AddToScheme(virtualScheme)
+	if err != nil {
+		fmt.Printf("unable to add client go types to virtual cluster scheme: %s", err.Error())
```

Comment:
```suggestion
		return fmt.Erorrf("unable to add client go types to virtual cluster scheme: %s", err.Error()")
```
You can just return the error here and the main will exit no need to call os.Exit(-1)

Reply:
My bad, I think that I wrote this back when this was the main method. I'll fix this and the other similar occurrences that I see.

Reply:
No worries, Thank you!